prototypes.js

```diff
@@ -0,0 +1,53 @@
+// Choose a topic/theme that you’re interested in.
+// For example Movies, Cars, Music, Sports etc.
+// (Don’t use the examples from the slides)
+
+// Implement the below prototypal inheritance diagram using the topic you chose
+// Create some instances with the child constructor functions and call the prototype methods
+
+function GenshinImpactCharacter(weapon, vision, rarity) {
+    this.weapon = weapon;
+    this.vision = vision;
+    this.rarity = rarity;
+    this.getConstellationNumber = function () {
+        return Math.ceil(Math.random() * 6);
+    }
+    this.getFriendshipLevel = function () {
+        return Math.ceil(Math.random() * 10)
+    }
+}
+
+function FemaleCharacter(weapon, vision, rarity) {
+    GenshinImpactCharacter.call(this, weapon, vision, rarity)
+    this.gender = 'female'
+}
+
+FemaleCharacter.prototype = Object.create(GenshinImpactCharacter.prototype)
```

Comment:
It is also important when you inherit also to reset the constructor.
FemaleCharacter.prototype.constructor = FemaleCharacter

And if you set the prototype of FemaleCharacter, you inherit getConstellationNumber and getFriendshipLevel if they are set on the prototype of GenshinImpactCharacter.